turbinia/workers/__init__.py

```diff
@@ -563,16 +566,19 @@ def execute(
       rw_paths = [self.output_dir, self.tmp_dir]
       container_manager = docker_manager.ContainerManager(docker_image)
       stdout, stderr, ret = container_manager.execute_container(
-          cmd, shell, ro_paths=ro_paths, rw_paths=rw_paths)
+          cmd, shell, ro_paths=ro_paths, rw_paths=rw_paths,
+          timeout_limit=timeout_limit)
 
     # Execute the job on the host system.
     else:
       if shell:
         proc = subprocess.Popen(
             cmd, shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+        proc.wait(timeout_limit)
```

Comment:
Can we add a check and a logline if this times out?  We should also make sure that the task gets closed with something like `success=False, status=<some error message about what timed out and how long it was running for>`.  It would also be nice to add an application metric for task timeouts.